roles/kubernetes-apps/ingress_controller/ingress_nginx/templates/ingressclass-nginx.yml.j2

```diff
@@ -0,0 +1,13 @@
+apiVersion: networking.k8s.io/v1
+kind: IngressClass
+metadata:
+  name: {{ ingress_nginx_class }}
+  labels:
+    app.kubernetes.io/name: ingress-nginx
+    app.kubernetes.io/part-of: ingress-nginx
+  {% if ingress_nginx_default %}
+  annotations:
+    ingressclass.kubernetes.io/is-default-class: "true"
+  {% endif %}
+spec:
+  controller: k8s.io/ingress-nginx
```

Comment:
The job was failed like
```
failed: [instance-1] (item={'diff': [], 'src': '/home/kubespray/.ansible/tmp/ansible-tmp-1684806141.5413666-21720-181580727079013/source', 'changed': True, 'group': 'root', 'uid': 0, 'dest': '/etc/kubernetes/addons/ingress_nginx/ingressclass-nginx.yml', 'checksum': '982fae9bffe3b581d34607213b52ccad8b7e66ea', 'md5sum': '8263d11d8ab1afeaae9b2e11e6507c27', 'owner': 'root', 'state': 'file', 'gid': 0, 'secontext': 'system_u:object_r:etc_t:s0', 'mode': '0644', 'invocation': {'module_args': {'src': '/home/kubespray/.ansible/tmp/ansible-tmp-1684806141.5413666-21720-181580727079013/source', 'directory_mode': None, 'force': True, 'unsafe_writes': False, 'remote_src': None, 'dest': '/etc/kubernetes/addons/ingress_nginx/ingressclass-nginx.yml', 'selevel': None, 'seuser': None, 'setype': None, 'group': None, 'content': None, '_original_basename': 'ingressclass-nginx.yml.j2', 'serole': None, 'mode': 420, 'checksum': '982fae9bffe3b581d34607213b52ccad8b7e66ea', 'owner': None, 'follow': False, 'validate': None, 'attributes': None, 'backup': False, 'local_follow': None}}, 'size': 215, 'failed': False, 'item': {'name': 'ingressclass-nginx', 'file': 'ingressclass-nginx.yml', 'type': 'ingressclass'}, 'ansible_loop_var': 'item'}) => {"ansible_loop_var": "item", "changed": false, "item": {"ansible_loop_var": "item", "changed": true, "checksum": "982fae9bffe3b581d34607213b52ccad8b7e66ea", "dest": "/etc/kubernetes/addons/ingress_nginx/ingressclass-nginx.yml", "diff": [], "failed": false, "gid": 0, "group": "root", "invocation": {"module_args": {"_original_basename": "ingressclass-nginx.yml.j2", "attributes": null, "backup": false, "checksum": "982fae9bffe3b581d34607213b52ccad8b7e66ea", "content": null, "dest": "/etc/kubernetes/addons/ingress_nginx/ingressclass-nginx.yml", "directory_mode": null, "follow": false, "force": true, "group": null, "local_follow": null, "mode": 420, "owner": null, "remote_src": null, "selevel": null, "serole": null, "setype": null, "seuser": null, "src": "/home/kubespray/.ansible/tmp/ansible-tmp-1684806141.5413666-21720-181580727079013/source", "unsafe_writes": false, "validate": null}}, "item": {"file": "ingressclass-nginx.yml", "name": "ingressclass-nginx", "type": "ingressclass"}, "md5sum": "8263d11d8ab1afeaae9b2e11e6507c27", "mode": "0644", "owner": "root", "secontext": "system_u:object_r:etc_t:s0", "size": 215, "src": "/home/kubespray/.ansible/tmp/ansible-tmp-1684806141.5413666-21720-181580727079013/source", "state": "file", "uid": 0}, 
"msg": "error running kubectl
 (/usr/local/bin/kubectl --namespace=ingress-nginx apply --force --filename=/etc/kubernetes/addons/ingress_nginx/ingressclass-nginx.yml)
 command (rc=1),
 out='',
 err='Error from server (BadRequest): error when creating \"/etc/kubernetes/addons/ingress_nginx/ingressclass-nginx.yml\":
 IngressClass in version \"v1\" cannot be handled as a IngressClass:
 strict decoding error: unknown field \"metadata.controller\", unknown field \"metadata.spec\"\n'"}
```

Reply:
Before the name was missing, but probably due to the same indentation issue, I hope this isnow fixed